File: languages/scala/exercises/concept/basics/LasagnaTest.scala
/** @version 1.3.0 */
class LasagnaTest extends FunSuite with Matchers {

  test("expected minutes in oven") {
    new Lasagna().expectedMinutesInOven() should be(40)
  }

  test("remaining minutes in oven after 25 minutes") {
    pending
    new Lasagna().remainingMinutesInOven(25) should be(15)
  }

  test("preparation time in minutes for one layer") {
    pending
    new Lasagna().preparationTimeInMinutes(1) should be(2)
  }

  test("preparation time in minutes for multiple layers") {
    pending
    new Lasagna().preparationTimeInMinutes(4) should be(8)
  }

  test("elapsed time in minutes for one layer") {
    pending
    new Lasagna().elapsedTimeInMinutes(1, 30) should be(32)
  }

  test("elapsed time in minutes for multiple layers") {
    pending
    new Lasagna().elapsedTimeInMinutes(4, 8) should be(16)
  }
}

File: languages/r/exercises/concept/lasagna/lasagna_test.R
library(testthat)

test_that("Expected minutes in the oven", {
  expect_equal(expected_minutes_in_oven(), 60)
})

test_that("Remaining minutes in the oven", {
  expect_equal(remaining_time_in_minutes(25), 35)
})

test_that("Preparation time in minutes for 1 layer", {
  expect_equal(prep_time_in_minutes(1), 2)
})

test_that("Preparation time in minutes for multiple layers", {
  expect_equal(prep_time_in_minutes(4), 8)
})

test_that("Elapsed time in minutes for 1 layer", {
  expect_equal(elapsed_time_in_minutes(1, 30), 32)
})

test_that("Elapsed time in minutes for multiple layers", {
  expect_equal(elapsed_time_in_minutes(4, 8), 16)
})
